code/datums/mutations/body.dm

```diff
@@ -394,3 +394,86 @@
 		return //remove the 'edge' cases
 	to_chat(owner, "<span class='danger'>You trip over your own feet.</span>")
 	owner.Knockdown(30)
+
+/datum/mutation/human/martyrdom
+	name = "Internal Martyrdom"
+	desc = "A mutation that makes the body destruct when near death. Not damaging, but very, VERY disorienting."
+	locked = TRUE
+	quality = POSITIVE //not that cloning will be an option a lot but generally lets keep this around i guess?
+	text_gain_indication = "<span class='warning'>You get an intense feeling of heartburn.</span>"
+	text_lose_indication = "<span class'notice'>Your internal organs feel at ease.</span>"
+
+/datum/mutation/human/martyrdom/on_acquiring()
+	RegisterSignal(owner, COMSIG_MOB_STATCHANGE, .proc/bloody_shower)
```

Comment:
Pretty sure both acquiring and losing need to call parent.

Reply:
var/mob/living/carbon/human/owner, we're good

Reply:
What?

Reply:
oh i misunderstood, will add this